osquery/remote/CMakeLists.txt

```diff
@@ -6,6 +6,11 @@ ADD_OSQUERY_LIBRARY(TRUE osquery_requests
   serializers/json.cpp
 )
 
+ADD_OSQUERY_LIBRARY(FALSE osquery_enrollment
+  enrollment/plugins/remote_enrollment.cpp
+)
+
 ADD_OSQUERY_TEST(TRUE requests_tests requests_tests.cpp)
 ADD_OSQUERY_TEST(TRUE json_tests serializers/json_tests.cpp)
 ADD_OSQUERY_TEST(TRUE http_tests transports/http_tests.cpp)
```

Comment:
I don't think we should have remote networking APIs in our public API, that's not really our thing. IMO, if we want to release those kinds of APIs, we should look around something like folly, clean our stuff up and submit it there or something.